contrib/debian/rules

```diff
@@ -12,10 +12,16 @@ override_dh_auto_clean:
 	if [ -f Makefile ]; then $(MAKE) distclean; fi
 	rm -rf Makefile.in aclocal.m4 configure src/Makefile.in src/bitcoin-config.h.in src/build-aux src/qt/Makefile.in src/qt/test/Makefile.in src/test/Makefile.in
 
+ifeq ($(shell dpkg-vendor --derives-from Ubuntu && echo yes),yes)
+	QT=qt4
+else
+	QT=qt5
+endif
+
 # Yea, autogen should be run on the source archive, but I like doing git archive
 override_dh_auto_configure:
 	./autogen.sh
-	./configure
+	./configure --without-miniupnpc --with-gui=$(QT)
```

Comment:
Don't forget to drop `--without-miniupnpc`?

(and restore `libminiupnpc8-dev | libminiupnpc-dev (>> 1.6),` in build deps)